components/core/CarouselSidebarData.js

```diff
@@ -338,11 +418,37 @@ export default class CarouselSidebarData extends React.Component {
     elements.push(
       <div key="s-2" css={STYLES_CONTAINER}>
         <div css={STYLES_META}>
-          <a css={STYLES_META_TITLE} target="_blank" href={Strings.getCIDGatewayURL(cid)}>
-            {file}
-          </a>
-          <div css={STYLES_META_DETAILS}>
-            <span css={STYLES_TAG}>{type}</span> <span>{Strings.bytesToSize(size)}</span>
+          {this.state.isEditing ? (
+            <Boundary enabled onOutsideRectEvent={this._handleEditFilename}>
+              <Input
+                full
+                value={this.state.name}
+                name="name"
+                onChange={this._handleChange}
+                id={`sidebar-label-name`}
+                style={STYLES_INPUT}
+              />
+            </Boundary>
+          ) : (
+            <span
+              css={STYLES_META_TITLE}
+              target="_blank"
+              // href={Strings.getCIDGatewayURL(cid)}
+              onClick={this._handleEditFilename}
+            >
+              {name}
```

Comment:
I'm still not seeing the filename change reflected immediately after editing. You can fix that by changing this line to {this.state.name} rather than name. That way this will always be the same value as the input field

Reply:
yeah I removed the debounce but there is still a bit lagging. thanks for pointing this out!